internal/controllers/addon/upgradepolicy_status.go

```diff
@@ -32,56 +35,81 @@ func (r *AddonReconciler) handleUpgradePolicyStatusReporting(
 		return nil
 	}
 
-	if addon.Spec.Version == "" {
-		return nil
-	}
-
-	if addon.Spec.UpgradePolicy == nil {
-		return nil
-	}
-
-	if addon.UpgradeCompleteForCurrentVersion() {
-		return nil
-	}
+	log = log.WithValues(
+		"UpgradePolicy", addon.Spec.UpgradePolicy,
+		"Version", addon.Spec.Version,
+	)
 
 	if addon.Status.UpgradePolicy == nil {
+		log.Info("UpgradePolicy status unknown; reporting upgrade as started")
```

Comment:
with the latest tests, I have the feeling we have to query the current state of the upgrade policy to decide what to do. We only report "started" if the current state is "scheduled", otherwise reporting will just fail.